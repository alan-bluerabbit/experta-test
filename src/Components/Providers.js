import * as React from 'react';
import Link from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import DeleteIcon from '@material-ui/icons/Delete';
import PersonIcon from '@material-ui/icons/Person';
import Title from './Title';
import Colors from '../Constants/Colors';

import MockData from '../Constants/mockData';

function preventDefault(event) {
  event.preventDefault();
}

const useStyles = makeStyles((theme) => ({
  seeMore: {
    marginTop: theme.spacing(3),
  },
}));

export default function Providers(props) {
  const classes = useStyles();
  return (
    <React.Fragment>
      <Title>Proveedores</Title>
      <Table size="small">
        <TableHead>
          <TableRow>
            <TableCell>Nombre</TableCell>
            <TableCell>CUIT</TableCell>
            <TableCell>Fecha de creación</TableCell>
            <TableCell align="right"></TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {MockData.map((row) => (
            <TableRow key={row.id}>
              <TableCell>{row.name}</TableCell>
              <TableCell>{row.cuit}</TableCell>
              <TableCell>{row.date}</TableCell>
              <TableCell align="right">
                <Button onClick={() => props.openProviderDialog(row)}>
                  <PersonIcon
                    style={{ color: Colors.primary.editColor }}
                  />
                </Button>
                <Button onClick={props.deleteProvider}>
                  <DeleteIcon
                    style={{ color: Colors.primary.editColor }}
                  />
                </Button>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
      <div className={classes.seeMore}>
        <Link color="primary" href="#" onClick={preventDefault}>
          Ver más proveedores
        </Link>
      </div>
    </React.Fragment>
  );
}