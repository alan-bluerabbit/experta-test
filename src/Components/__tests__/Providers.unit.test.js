import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '../Providers';

test('<Providers />', () => {
    const providers = [
        {
            id: 123456,
            name: 'testName',
            cuit: '20-12345678-7',
            created: 1234567,
        }
    ]
    const tree = renderer
    .create(<Providers providers={providers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});