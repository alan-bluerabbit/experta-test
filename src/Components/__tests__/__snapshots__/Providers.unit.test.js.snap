// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Providers /> 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Proveedores
  </h2>,
  <table
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Nombre
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          CUIT
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Fecha de creación
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          testName
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          20-12345678-7
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Thu Jan 15 1970
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "color": "grey",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "color": "grey",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
          </button>
        </td>
      </tr>
    </tbody>
  </table>,
  <div
    className="makeStyles-seeMore-1"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="#"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Ver más proveedores
    </a>
  </div>,
]
`;
